src/dvclive/data/plot.py

```diff
@@ -6,7 +6,7 @@
 
 class Plot(Data):
     suffixes = [".json"]
-    subfolder = "plots"
+    subfolder = "data_series"
```

Comment:
I believe this is the only name I don't really like. It doesn't really match the meaning of VSCode extension as their `data series` is our `metrics` and this folder can contain things like confusion matrix, so it doesn't really apply

Reply:
`sklearn` seems too implementation-focused, but it's mentioned in the [docs](https://dvc.org/doc/dvclive/api-reference/live/log_plot), and it's more descriptive than `data_series`.

Reply:
`json` is another that's too implementation-focused but at least descriptive.

Reply:
Or maybe something like `builtin` or `presets`?

Reply:
@shcheklein @jorgeorpinel 
Any suggestions regarding alternatives to the `data_series` name (and/or the rest of the outputs)? 

`data_series` is meant to store pre-defined plots based on [scikit-learn visualizations ](https://scikit-learn.org/stable/visualizations.html).

The current state is:

| Method  | Output |
| ------------- | ------------- |
| [live.log](https://dvc.org/doc/dvclive/api-reference/live/log) | `dvclive/plots/metrics` & `dvclive/metrics.json` |
| [live.log_image](https://dvc.org/doc/dvclive/api-reference/live/log_image)  | `dvclive/plots/images` |
| [live.log_param](https://dvc.org/doc/dvclive/api-reference/live/log_param)  | `dvclive/params.yaml` |
| [live.log_plot](https://dvc.org/doc/dvclive/api-reference/live/log_lot)  | `dvclive/plots/data_series` |

Reply:
Just checked in the VSCode extension that `Data Series` actually contains any template (including confusion matrix) so I guess we are actually being consistent here

Reply:
yep, to be honest I also don't like it (had the same struggle when I was trying to come with VS Code section names). Some ideas from the top of my head:

- would it make sense to try to make `live.log` take care of `live.log_plot`. A bit of confusion / ugliness come from the mismatch between `data_series` and `_plot` in the name.
- alternative could be to dump plots right under the `dvclive/plots` (+ still have `images` and `metrics` as special subfolders)?

Reply:
> * would it make sense to try to make `live.log` take care of `live.log_plot`

Instead of combining the methods, we could rename `live.log_plot` to match whatever we decide for a name here.

Maybe a good convention is to name the methods according to their inputs (rather than thinking about the outputs), since it:
1. Is more self-explanatory to the user what to input.
2. Leaves flexibility to change the outputs without breaking the method naming. 
 
With that in mind, I think `live.log_sklearn` makes sense since the inputs closely follow the underlying sklearn methods. The output could then be `plots/sklearn`.

Reply:
(btw, by saying "I don't like it" I don't mean that we should not go with this - I'm sharing my thoughts. If we don't come with something better we should proceed with this)

> With that in mind, I think live.log_sklearn makes sense since the inputs closely follow the underlying sklearn methods. The output could then be plots/sklearn.

hmm, that's interesting ... maybe I missed something, but why `sklearn` specifically?

Reply:
From https://dvc.org/doc/dvclive/api-reference/live/log_plot and the link at the top (which may need to be updated because it doesn't link to the expected methods), the method arguments come from sklearn. Most sklearn evaluation methods take similar arguments like `metrics.method(y_true, y_score)`. 


If we start to log any other plot types that take different arguments, we are likely to need a separate method. 

For users who are familiar, it's the easiest way to explain the inputs, and sklearn is used for evaluation even when it's not used for modeling (the top result for pytorch roc curve is https://discuss.pytorch.org/t/how-to-plot-roc-curve-using-pytorch-model/122033).

Reply:
> From https://dvc.org/doc/dvclive/api-reference/live/log_plot and the link at the top ...

Got it, thanks! I forgot (not the first time :) ) that interface there was coupled with the sklearn types of plots. Yes, it makes more sense to rename into `sklearn_plot`. On the other hand, we already do things like this:

https://github.com/iterative/example-get-started/blob/main/src/evaluate.py#L49-L59

should it be taken care of by the some kind of `log_plot("<str>") ?

feels like even if rename the existing one to `log_sklearn_plot` or something, we'll need the second `log_plot` pretty soon (now?)

Reply:
>  the method arguments come from sklearn. Most sklearn evaluation methods take similar arguments like 

Not only that. We use `sklearn` internally to compute the results to be plotted.

> > If we start to log any other plot types that take different arguments, we are likely to need a separate method.

Indeed, we would need to define a different interface for plots beyond the predefined `sklearn` set. 

What `wandb` does is provide [secondary methods](https://docs.wandb.ai/guides/track/log/plots) that are passed as arguments to `wandb.log`:

```python
data = [[x, y] for (x, y) in zip(x_values, y_values)]
table = wandb.Table(data=data, columns = ["x", "y"])
wandb.log({"my_custom_plot_id" : wandb.plot.line(table, "x", "y",
           title="Custom Y vs X Line Plot")})
```

> should it be taken care of by the some kind of `log_plot("") ?

It could already use `log_sklearn_plot("precision_recall", ...)` but afaik (per the comment above) it was kept to showcase an alternative approach of manually generating a plot file.

Reply:
Sounds like we can move forward with `log_sklearn_plot`?

> > should it be taken care of by the some kind of `log_plot("") ?
> 
> It could already use `log_sklearn_plot("precision_recall", ...)` but afaik (per the comment above) it was kept to showcase an alternative approach of manually generating a plot file.

Even if we don't need it for the example get started repo, it would probably make sense after https://github.com/iterative/dvclive/issues/311 to introduce a more generic method that logs any dvc plot (`log_custom_plot`?). Should we discuss in #271?